File: find-greatest-common-divisor-of-array/find-greatest-common-divisor-of-array.java
class Solution {
    public int findGCD(int[] nums) {
        
        int nMin = 1001, nMax = 0, remainder;
        
        for (int n : nums) {
            
            nMin = Math.min(nMin, n);
            nMax = Math.max(nMax, n);
        }
        
        while (nMin > 0) {
            
            remainder = nMax % nMin;
            
            nMax = nMin;
            
            nMin = remainder;
        }
        
        return nMax;
    }
}

File: find-greatest-common-divisor-of-array/find-greatest-common-divisor-of-array.cpp
class Solution {
public:
    int findGCD(vector<int>& nums) {
        
        int nMin = *std::min_element(nums.begin(), nums.end()), 
            nMax = *std::max_element(nums.begin(), nums.end());
        
        return std::gcd(nMin, nMax);
    }
};